pymongo/database.py

```diff
@@ -904,19 +914,33 @@ def profiling_level(self, session=None):
           - `session` (optional): a
             :class:`~pymongo.client_session.ClientSession`.
 
+        .. versionchanged:: 3.12
+           Deprecated.
+
         .. versionchanged:: 3.6
            Added ``session`` parameter.
 
         .. mongodoc:: profiling
+        .. _profile command: https://docs.mongodb.com/manual/reference/command/profile/
         """
+        warnings.warn("profiling_level() is deprecated. See the documentation "
+                      "for more information",
+                      DeprecationWarning, stacklevel=2)
         result = self.command("profile", -1, session=session)
 
         assert result["was"] >= 0 and result["was"] <= 2
         return result["was"]
 
     def set_profiling_level(self, level, slow_ms=None, session=None,
                             sample_rate=None, filter=None):
-        """Set the database's profiling level.
+        """**DEPRECATED**: Set the database's profiling level.
+
+        Starting with PyMongo 3.12, this helper is obsolete. The functionality
+        provided by this helper is available in MongoDB 3.6+ using the
+        `profile command`_, which can be run using the :meth:`command`
+        helper, e.g.::
+
+           profile_info = db.command({'profile': 2, 'filter': {'op': 'query'}})
```

Comment:
I'm ambivalent about removing the constants. They don't cost anything to keep but if you want remove they we start by saying they are deprecated in the docstring like we did for pymongo.GEOHAYSTACK in PYTHON-2185.

I also suggest using SON or this command-style invocation to be compatible with unordered dict:
 ```
 res = db.command('profile', 2, filter={'op': 'query'})
 ```

Reply:
Deprecated the constants mostly because there might be more added in the future so why burden ourselves with maintaining those if the user has to use the raw command helper anyway?

Also changed the examples to use command-style invocation.